### PR TITLE
Add Stripe.js V3 with Elements

### PR DESCRIPTION
**Pull request by spaghetticode**
Ref #5

Besides the old Stripe.js V2 API, it's now possible to use the recent Stripe.js V3 library with Elements. 

A new model preference was added to `Spree::PaymentMethod::StripeCreditCard` allowing to choose
between the two different libraries in the admin area: 

Schermata 2019-04-29 alle 16 40 00

The new combination of Stripe.js V3 with Elements allows PCI SAQ-A certification eligibility.

### CODE REVIEW

**Comment by jackie577** [comment]
Hi there, thanks for working on this! Would love to know if/when you are planning to release this. (:

**Comment by spaghetticode** [comment]
@jackie577 now that specs are green I think the maintainers will review the code, hopefully it will get their approval and be merged soon.

File: lib/generators/solidus_stripe/install/install_generator.rb
Comment (jacobherrington) [inline lib/generators/solidus_stripe/install/install_generator.rb]: Is there a scenario where someone uses this library and `vendor/assets/stylesheets/spree/frontend/all.css` isn't present? I believe this will throw an error. Will this work if someone tries to use this library with a custom frontend?

**Comment by jacobherrington** [comment]
Thanks @spaghetticode, this looks great. I have one question, but otherwise, I'm happy with the PR.

File: lib/generators/solidus_stripe/install/install_generator.rb
Comment (spaghetticode) [reply lib/generators/solidus_stripe/install/install_generator.rb]: @jacobherrington you're right. I tested it right now by changing the name of the file to a non-existing path, such as 

```inject_into_file "vendor/fake/frontend/fake.css", " *= require spree/frontend/solidus_stripe\n", before: '*/', verbose: true```

and Thor throws an error. 

What about adding an `if` statement that checks the existence of the file? I think this works fine: 

``` 
filename = 'vendor/assets/stylesheets/spree/frontend/all.css'
if File.file? filename
  inject_into_file filename, " *= require spree/frontend/solidus_stripe\n", before: '*/', verbose: true
end
```

I already added a commit to verify it works on Travis too.

Not sure I understand what you mean with the last question. The added styles are just cosmetics, and everything works fine also without them. Stripe.js V3 makes some assumptions regarding the HTML and the JS behavior, see their [tutorial](https://stripe.com/docs/stripe-js/elements/quickstart). As long as the custom frontend respects these assumptions then it's good to go.

**Review by jacobherrington — APPROVED** [review]
LGTM. Good work @spaghetticode!

**Pull request merged**
